libs/core/src/lib/message-toast/message-toast.component.ts

```diff
@@ -0,0 +1,226 @@
+    ElementRef,
+    EmbeddedViewRef,
+    EventEmitter,
+    HostBinding,
+    HostListener,
+    Input,
+    NgZone,
+    OnInit,
+    Optional,
+    Output,
+    TemplateRef,
+    Type,
+    ViewChild,
+    ViewContainerRef,
+    ViewEncapsulation
+} from '@angular/core';
+import { MessageToastConfig } from './message-toast-utils/message-toast-config';
+import { MessageToastRef } from './message-toast-utils/message-toast-ref';
+
+let messageToastUniqueId = 0;
+
+/**
+ * The component that represents a message toast.
+ */
+@Component({
+    selector: 'fd-message-toast',
+    template: `<ng-container #container>{{ message }}</ng-container>`,
+    styleUrls: ['./message-toast.component.scss'],
+    host: {
+        '[attr.aria-label]': 'ariaLabel',
+        '[attr.id]': 'id',
+        '[style.width]': 'width',
+        '[style.min-width]': 'minWidth',
+        '[style.max-width]': 'maxWidth'
+    },
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MessageToastComponent implements OnInit, AfterViewInit {
+    /** @hidden */
+    constructor(
+        private elRef: ElementRef,
+        private cdRef: ChangeDetectorRef,
+        private componentFactoryResolver: ComponentFactoryResolver,
+        private ngZone: NgZone,
+        @Optional() private messageToastConfig: MessageToastConfig,
+        @Optional() private messageToastRef: MessageToastRef
+    ) {
+        this._setMessageToastConfig(messageToastConfig);
+    }
+
+    /** @hidden */
+    @ViewChild('container', { read: ViewContainerRef })
+    containerRef: ViewContainerRef;
+
+    /** Id of the message toast. If omitted, a unique one is generated. */
+    @Input()
+    id: string = 'fd-message-toast-' + messageToastUniqueId++;
+
+    /**
+     * Duration of time *in milliseconds* that the message toast will be visible.
+     * The standard value is 3000 ms.
+     * The duration can be set to more than 3000 ms,
+     * but it's not recommended to set it to less than 3000 ms.
+     * Set to -1 for indefinite.
+     */
+    @Input()
+    duration = 3000;
+
+    /** Whether the message toast should stay open if the cursor is over it. */
+    @Input()
+    mousePersist = false;
+
+    /** Aria label for the message toast component element. */
+    @Input()
+    ariaLabel: string = null;
+
+    /** Alternative way of passing in a message to the message toast. */
+    @Input()
+    message: string;
+
+    /** Width of the message toast. */
+    @Input()
+    width: string;
+
+    /** Minimum width of the message toast. */
+    @Input()
+    minWidth: string;
+
+    /**
+     * Maximum width of the message toast.
+     * The standard maximum width is 15rem.
+     * It's not recommended to extend the width to more than 35rem.
+     */
+    @Input()
+    maxWidth: string;
+
+    /** Event fired when the message toast is timeout. */
+    @Output()
+    onTimeout: EventEmitter<undefined> = new EventEmitter<undefined>();
```

Comment:
It would be better to set the name 'onClose' or 'onHide' because you fire this event on a close only.